src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -210,6 +210,15 @@ private async Task RegisterDesignerAttributeAsync(Document document, string desi
                 return;
             }
 
+            var hierarchy = workspace.GetHierarchy(document.Project.Id);
+            if (hierarchy == null)
+            {
+                // make sure the project still exists in VS.
+                // since we are running under async call, it is possible that when we run
+                // project is already removed from VS.
+                return;
+            }
```

Comment:
It looks like this is called from a free-affinitized location.  i.e. RegisterDesignerAttributeAsync is called after a `.ConfigureAwait(False)`

So this seems racey.  You could call this, get a hierarchy back, but then have the hierarchy gone immediately after getting to this point.